src/ReaderMiddleware.ts

```diff
@@ -750,31 +750,39 @@ export const altW: <R2, I, E2, A>(
 
 /**
  * @category combinators
- * @since 0.6.3
+ * @since 0.7.9
  */
-export const chainMiddlewareK =
-  <R, I, E, A, B>(f: (a: A) => M.Middleware<I, I, E, B>) =>
-  (ma: ReaderMiddleware<R, I, I, E, A>): ReaderMiddleware<R, I, I, E, B> =>
-    pipe(
-      ma,
-      chain((a) => fromMiddleware(f(a)))
-    )
+export const fromMiddlewareK =
+  <R, A extends ReadonlyArray<unknown>, B, I, O, E>(
+    f: (...a: A) => M.Middleware<I, O, E, B>
+  ): ((...a: A) => ReaderMiddleware<R, I, O, E, B>) =>
+  (...a) =>
+    fromMiddleware(f(...a))
+
+/**
+ * @category combinators
+ * @since 0.6.5
+ */
+export const ichainMiddlewareKW =
+  <R, A, O, Z, E, B>(f: (a: A) => M.Middleware<O, Z, E, B>) =>
+  <I, D>(ma: ReaderMiddleware<R, I, O, D, A>): ReaderMiddleware<R, I, Z, D | E, B> =>
+    pipe(ma, ichainW(fromMiddlewareK(f)))
 
 /**
  * @category combinators
  * @since 0.6.3
  */
 export const ichainMiddlewareK: <R, A, O, Z, E, B>(
   f: (a: A) => M.Middleware<O, Z, E, B>
-) => <I>(ma: ReaderMiddleware<R, I, O, E, A>) => ReaderMiddleware<R, I, Z, E, B> = chainMiddlewareK as any
+) => <I>(ma: ReaderMiddleware<R, I, O, E, A>) => ReaderMiddleware<R, I, Z, E, B> = ichainMiddlewareKW
 
 /**
  * @category combinators
- * @since 0.6.5
+ * @since 0.6.3
```

Comment:
```suggestion
 * @since 0.7.9
```

Reply:
The diff's a bit confusing as the order changes; the resulting docs show the only change is the addition of `fromMiddlewareK` in `0.7.9`.